assignments/array-methods.js

```diff
@@ -54,23 +54,55 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 {"id":50,"first_name":"Shell","last_name":"Baine","email":"sbaine1d@intel.com","shirt_size":"M","company_name":"Gabtype","donation":171}];
 
 // ==== Challenge 1: Use .forEach() ====
-// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
+// The event director needs both the first and last names of each runner for their running 
+// bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+runners.forEach( function(currentValue) { 
+    fullName.push(currentValue.first_name); 
+    fullName.push(currentValue.last_name); 
+  });
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
+const allCap = runners.map( first_name => first_name.toUpperCase);
+  allCaps.push(allCap.first_name)
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
-console.log(largeShirts);
+
+// for(let i = 0; i < runners.length; i++) {
+//     function largeRunners() {
+//         if( runners[i].shirt_size === "L") {
+//             largeShirts.push(runners[i].shirt_size)
+//         }
+//     }
+// }
+
+const largeRunners = runners.filter( function(currentValue){
+    if( currentValue.shirt_size === "L") {
+        largeShirts.push(currentValue.shirt_size)
+    }
+  });
```

Comment:
no need to push here in filter. filter will return an array with the items that pass your requirements. forEach does not return an array, so that's why we had to push from that one. in this case you can just set largeShirts equal to your filter. you also don't even need an if. filter will understand if you say `largeShirts = runners.filter( function(currentValue){return currentValue.shirt_size === 'L'}`, that tells it to return (and save to an array) the runners with shirts size large